common/src/main/java/com/wynntils/features/chat/InputTranscriptionFeature.java

```diff
@@ -0,0 +1,242 @@
+
+        if (selectedAlphabet == WynnAlphabet.DEFAULT) return;
+
+        // Can't use parseInt here as '０', '１', '２' are used for '.', '!', '?' in Wynnic
+        Matcher numMatcher = NUMBER_PATTERN.matcher(event.getTextToWrite());
+
+        if (numMatcher.matches()) {
+            if (selectedAlphabet != WynnAlphabet.WYNNIC) return;
+
+            handleTypedNumber(
+                    Integer.parseInt(event.getTextToWrite()),
+                    Models.WynnAlphabet.getWynnicNumbers(),
+                    event,
+                    chatScreen);
+        } else {
+            List<Character> replacementList = selectedAlphabet == WynnAlphabet.GAVELLIAN
+                    ? Models.WynnAlphabet.getGavellianCharacters()
+                    : Models.WynnAlphabet.getWynnicCharacters();
+
+            handleTypedCharacter(event.getTextToWrite(), replacementList, event, chatScreen);
+        }
+    }
+
+    @SubscribeEvent
+    public void onChatScreenKeyTyped(ChatScreenKeyTypedEvent event) {
+        if (event.getKeyCode() != GLFW.GLFW_KEY_BACKSPACE) return;
+        if (!(McUtils.mc().screen instanceof ChatScreen chatScreen)) return;
+        if (!transcriptionButtons.get()) return;
+        if (!chatScreen.input.getHighlighted().isBlank()) return;
+
+        String beforeCursor = chatScreen.input.getValue().substring(0, chatScreen.input.getCursorPosition());
+        String afterCursor = chatScreen.input.getValue().substring(chatScreen.input.getCursorPosition());
+
+        if (beforeCursor.isBlank()
+                || !Models.WynnAlphabet.getWynnicNumbers().contains(beforeCursor.charAt(beforeCursor.length() - 1)))
+            return;
+
+        String currentNumsStr = getWynnicNumBeforeCursor(beforeCursor);
+
+        int wynnicNum = Models.WynnAlphabet.wynnicNumToInt(currentNumsStr);
+
+        String backspacedNum = String.valueOf(wynnicNum);
+
+        if (backspacedNum.length() > 1) {
+            event.setCanceled(true);
+
+            backspacedNum = backspacedNum.substring(0, backspacedNum.length() - 1);
+
+            int index = beforeCursor.lastIndexOf(currentNumsStr);
+
+            beforeCursor = beforeCursor.substring(0, index);
+
+            updateInput(beforeCursor, afterCursor, Integer.parseInt(backspacedNum), chatScreen.input);
+        }
+    }
+
+    @SubscribeEvent
+    public void onChatSend(ChatSentEvent event) {
+        if (event.getMessage().isBlank()) return;
+
+        String message = event.getMessage();
+
+        if (Models.WynnAlphabet.containsBrackets(message)) {
+            String updatedMessage = Models.WynnAlphabet.transcribeBracketedText(message);
+
+            updatedMessage = updatedMessage.substring(0, Math.min(updatedMessage.length(), MAX_CHAT_LENGTH));
+
+            if (!updatedMessage.equals(message)) {
+                event.setCanceled(true);
+                McUtils.mc().getConnection().sendChat(updatedMessage);
```

Comment:
I just pushed McUtils.sendChat() to main. The asymmetry with commands here killed me :)